libp2p/netMessenger.go

```diff
@@ -1000,7 +1011,7 @@ func (netMes *networkMessenger) RegisterMessageProcessor(topic string, identifie
 		topicProcs = newTopicProcessors()
 		netMes.processors[topic] = topicProcs
 
-		err := netMes.registerOnPubSub(topic, topicProcs)
+		err := netMes.pb.RegisterTopicValidator(topic, netMes.pubsubCallback(topicProcs, topic))
```

Comment:
On elrond-go, you kept the private method registerOnPubSub. Indeed the current implementation is the same since it just returns netMes.pb.RegisterTopicValidator(topic, netMes.pubsubCallback(topicProcs, topic)).
Maybe you can add a todo task, to remove this method also from elrond-go.